Subscribie/subscribie
# 293 stripe subscriptions (GoCardless removal)

- #293 removal of GoCardless and replace with Stripe for subscriptions

- Also introduces playwright browser automation tests (currently 4 browser tests) https://github.com/Subscribie/subscribie/blob/c13ddf80d441d40162a9ea09da1e7061d86d5263/tests/browser-automated-tests-playwright/index.js#L41

## How to QA this (manually)

Visit 
http://chrisjsimpson-293-stripe-subscriptions.pcpink.co.uk/

And login as a shop owner using test credentials: 
Username: admin@example.com
Password: 
password

- Complete the 'connect to stripe' flow, using your email as a test account (you may skip sms confirmation in test mode).
- Complete the sign-up to a plan, as a customer. 
- Verify Transaction appears, new subscriber, and an invoice is present
- Complete the sign-up of a different plan

- See tests for an an idea of different payment scenarios: https://github.com/Subscribie/subscribie/blob/293-stripe-subscriptions/tests/browser-automated-tests-playwright/index.js#L41


## Screens

The following screenshots are taken using the playwright test automation for one of the tests (there are currently 4 playwright tests). There's only one shown below

### Test name: test_order_plan_with_subscription_and_upfront_charge
#### New customer form
![new-customer-form-webkit](https://user-images.githubusercontent.com/1718624/101837931-ba505900-3b37-11eb-98d5-3940e91fa383.png)

#### Pre stripe checkout 
![pre-stripe-checkout-webkit](https://user-images.githubusercontent.com/1718624/101837930-b9b7c280-3b37-11eb-9151-e5080b35d3a1.png)

#### Stripe checkout filled in
![stripe-checkout-filled-in-webkit](https://user-images.githubusercontent.com/1718624/101837944-bb818600-3b37-11eb-850f-fc5a1a9e5b0d.png)

#### Order complete
![order-complete-webkit](https://user-images.githubusercontent.com/1718624/101837940-bb818600-3b37-11eb-9c8e-82d7624d85f7.png)

#### Logged in (shop owner) to check order

no image

#### View subscriber

![view-subscribers-webkit](https://user-images.githubusercontent.com/1718624/101837936-ba505900-3b37-11eb-9cd8-1c841396c727.png)

#### View transactions

![view-transactions-chromium](https://user-images.githubusercontent.com/1718624/101837925-b8869580-3b37-11eb-9977-b697ac2f7a09.png)

#### view-paid-invoices

![view-paid-invoices-chromium](https://user-images.githubusercontent.com/1718624/101837995-cc31fc00-3b37-11eb-9a25-daf647815967.png)

#### view-upcoming-invoices

![view-upcoming-invoices-chromium](https://user-images.githubusercontent.com/1718624/101837994-cc31fc00-3b37-11eb-8147-5bd2a0cd41d1.png)

![display-stripe-checkout-chromium](https://user-images.githubusercontent.com/1718624/101837929-b9b7c280-3b37-11eb-9381-e2b9365b5a75.png)

#### Logged-out

![logged-out-webkit](https://user-images.githubusercontent.com/1718624/101837937-bae8ef80-3b37-11eb-8bd9-470a64e1b660.png)
